scripts/deploy.sh

```diff
@@ -18,6 +18,19 @@ npm run build
 cp package.json dist
 setup_git
 
+echo "Dry-run of semantic-version to determine whether the package will be published"
+# merge stderr and stdout and check for "no new version is released". If found, bail.
+# don't forget to capture the grep failure with `|| echo ""` so that the script doesn't fail
+# when the line isn't found
+npm run semantic-release -- -d 2>&1 | grep "no new version is released" > no_new_version.txt || echo "It looks like a new version will be released!"
+if [ -s no_new_version.txt ]; then
+    # run the dry run a second time so that the output is printed to the console
+    echo npm run semantic-release -- -d
```

Comment:
this is just wastefull IMO, lets simply:
`npm run semantic-release -- -d &> sematic_out.txt`
and then grep on sematic_out.txt and cat its value

Reply:
that's fair. it won't be as pretty, but it'll be fine like that

Reply:
actually, i like your solution better, i used `tee` and it's much cleaner 🙏